services/blockstorage/sync/state_finished_car.go

```diff
@@ -0,0 +1,47 @@
+package sync
+
+import (
+	"context"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/gossipmessages"
+	"math/rand"
+)
+
+type finishedCARState struct {
+	responses []*gossipmessages.BlockAvailabilityResponseMessage
+	logger    log.BasicLogger
+	sf        *stateFactory
+}
+
+func (s *finishedCARState) name() string {
+	return "finished-collecting-availability-requests-state"
+}
+
+func (s *finishedCARState) processState(ctx context.Context) syncState {
+	if ctx.Err() == context.Canceled {
```

Comment:
eh?

Reply:
checking if context was terminated (thats the way to do it without blocking on the channel)